code/game/objects/items/weapons/weaponry.dm

```diff
@@ -460,8 +460,33 @@ obj/item/weapon/banhammer/admin
 			to_add.pixel_x -= 12 * PIXEL_MULTIPLIER
 	to_add.plane = FLOAT_PLANE
 	underlays += to_add.appearance
+
+/obj/item/weapon/proc/pick_rand_tele_turf(atom/hit_atom, var/inner_teleport_radius, var/outer_teleport_radius)
+	if((inner_teleport_radius < 1) || (outer_teleport_radius < inner_teleport_radius))
+		return 0
+
+	var/list/turfs = new/list()
+	var/turf/hit_turf = get_turf(hit_atom)
+	//This could likely use some standardization but I have no idea how to not break it.
+	for(var/turf/T in trange(outer_teleport_radius, hit_turf))
+		if(get_dist(T, hit_atom) <= inner_teleport_radius)
+			continue
+		if(is_blocked_turf(T) || istype(T, /turf/space))
+			continue
+		if(T.x > world.maxx-outer_teleport_radius || T.x < outer_teleport_radius)
+			continue
+		if(T.y > world.maxy-outer_teleport_radius || T.y < outer_teleport_radius)
+			continue
+		turfs += T
+	if(!turfs.len)
+		var/list/turfs_to_pick_from = list()
+		for(var/turf/T in trange(outer_teleport_radius, get_turf(hit_atom)))
+			if(get_dist(T, hit_atom) > inner_teleport_radius)
+				turfs_to_pick_from += T
+		turfs += pick(/turf in turfs_to_pick_from)
```

Comment:
I know it's copypasted but does this work?
Also why is this an `/obj/item` proc?

Reply:
Bluespace tomatoes are /obj/item/weapon/reagent_containers/etc/etc, whereas peels are /obj/item/weapon, so I shuffled the proc up from reagent_containers/etc to weapon.

Reply:
Why not just a top level proc?